### ISSUE TITLE
Decide on project license

### ISSUE THREAD

**Issue by johanbove** [jqPlot/jqPlot#176]
The current copyright of this project looks like this:

> jqPlot is currently available for use in all personal or commercial projects under both the MIT (http://www.opensource.org/licenses/mit-license.php) and GPL version 2.0 (http://www.gnu.org/licenses/gpl-2.0.html) licenses. This means that you can choose the license that best suits your project and use it accordingly. 

I'm not much into legal topics, but wouldn't it be better if we kept just a single license instead offering this "choice" ? What's the point of this anyhow?

**Comment by rswestmoreland** [comment]
I agree, MIT is the way to go to keep the project completely available to
use however anyone sees fit.

**Comment by toolstack** [comment]
Dual licencing isn't a bad thing and changing it now is more problematic as you would have to go to each contributor and get permission to change the licence terms.  Each contributor still holds the copyright on the code they submitted.

**Comment by rswestmoreland** [comment]
Wouldn't it be fine to change the license on a major version change?  So
all releases up to that version are dual, then the new version is just MIT.

**Comment by toolstack** [comment]
No, that's not how licencing works, since the contributors submitted the code with a dual licence, to change it you would still have to get permission from each individual contributor, irregardless of if it's a minor or major release.

> Wouldn't it be fine to change the license on a major version change? So all releases up to that version are dual, then the new version is just MIT.
> […](#)
> On Fri, Sep 28, 2018 at 12:03 PM Greg Ross ***@***.***> wrote: Dual licencing isn't a bad thing and changing it now is more problematic as you would have to go to each contributor and get permission to change the licence terms. Each contributor still holds the copyright on the code they submitted.

**Comment by johanbove** [comment]
As @toolstack pointed out correctly, we'll need to consult with all contributors. But that shouldn't be too hard as there aren't that many people involved (yet).

Also picking a single license will make jqPlot easier to be found on GitHub through "[search by licence](https://help.github.com/articles/searching-for-repositories/#search-by-license)"; see also https://help.github.com/articles/licensing-a-repository/

I found a good resource, by GitHub, on what changing a license in open source projects entails; see https://opensource.guide/legal/#what-if-i-want-to-change-the-license-of-my-project

This project has at the moment 34 Contributors from 2009 until now.

Asking all contributors if they agree that we **switch jqPlot to MIT license** which is easier for everyone and it allows using jqPlot in closed source projects. (cf. https://choosealicense.com/);

**Please add your comment, objection, reaction to this thread if you agree or disagree.**

## Contributors with more than one commit:

@cleonello 
@ppritcha 
@akuchling 
@johanbove 
@harbulot 
@KTW-NIRAS 
@spoonguard2k 
@jeschr 
@tomascassidy 
@harryzhux 
@DaltonNotetech 
@audriusk 
@svenjacobs 

## Contributors with 1 commit:

@dwhipps 
@jasonex7 
@Coeur 
@clancelotti 
@walterbrebels 
@BradPenwarden 
@gkjothi20 
@psaliente 
@toolstack 
@pc-m 
@simonschaufi 
@y2chen 
@Amomo 
@FO-nTTaX 
@piyushsaini123 
@veger 
@dg-spark 
@hkirk 
@BBBThunda 
@pchop2 
@arthurlogilab 

@JordiCorbilla 

Thank you all!

**Comment by simonschaufi** [comment]
I'm fine with a change.

**Comment by ktw** [comment]
@KTW-NIRAS is my work account, so fine with me.

**Comment by svenjacobs** [comment]
I'm fine with the MIT license, too.

**Comment by JordiCorbilla** [comment]
I agree too!

**Comment by pc-m** [comment]
MIT is fine for me.

**Comment by harbulot** [comment]
I'm fine with the MIT license.

(That said, I'm not entirely sure it's worth the trouble. Once it's MIT-only, it could cause problems for people who use it as GPL-only. I'm not sure how to assess the legal arguments of the answers to [this question on StackExchange](https://softwareengineering.stackexchange.com/questions/105912/can-you-change-code-distributed-under-the-mit-license-and-re-distribute-it-unde), but it raises some interesting points.)

**Comment by audriusk** [comment]
I'm OK with the change to MIT-only license.

**Comment by FO-nTTaX** [comment]
MIT is fine with me, I only fixed some typos anyways.

**Comment by dwhipps** [comment]
Yup. Fine.

**Comment by toolstack** [comment]
I'm fine with MIT.

**Comment by akuchling** [comment]
I am also fine with the change to MIT, though I don't see the necessity of dropping the GPL.

1.

**Comment by Coeur** [comment]
I grant all permissions for any licence change.

**Comment by veger** [comment]
Changing license is fine by me

**Comment by walterbrebels** [comment]
I grant all permissions for any licence change.

Op zo 30 sep. 2018 om 09:57 schreef Maarten Bezemer <
notifications@github.com>:

**Comment by y2chen** [comment]
MIT is fine.

**Comment by DaltonNotetech** [comment]
MIT licence will be fine.
I think, GPL forces the release of source code, which most commercial projects wont like.

**Comment by piyushsaini123** [comment]
I am fine with MIT license.

**Comment by jasonex7** [comment]
I agree with the MIT license

**Comment by tomascassidy** [comment]
License change is OK with me

**Comment by harryzhux** [comment]
I agree using the MIT license.

**Comment by arthurlogilab** [comment]
OK for me too

**Comment by dg-spark** [comment]
I grant all permissions for any licence change.

**Comment by BBBThunda** [comment]
If you're still looking for feedback I have no objections to changing the license.

**Comment by johanbove** [comment]
Thanks to all of you who gave their response to this - long running - poll.

The results as of today are:

| Vote| Count |
| --- | --- |
| 😃 | 35 |
| 👍 | 24 |
| 👎 | 0 |
| ❓ | 11 |

## Contributors with more than one commit:

@cleonello ❓ 
@ppritcha  ❓ 
@akuchling 👍 
@johanbove 👍 
@harbulot 👍 
@KTW-NIRAS 👍 
@spoonguard2k 👍
@jeschr  ❓ 
@tomascassidy 👍
@harryzhux 👍
@DaltonNotetech 👍 
@audriusk 👍 
@svenjacobs 👍 

## Contributors with 1 commit:

@dwhipps 👍 
@jasonex7 👍
@Coeur 👍 
@clancelotti  ❓ 
@walterbrebels  ❓ 
@BradPenwarden  ❓ 
@gkjothi20  ❓ 
@psaliente 👍
@toolstack 👍 
@pc-m 👍 
@simonschaufi 👍
@y2chen 👍 
@Amomo  ❓ 
@FO-nTTaX 👍 
@piyushsaini123 👍 
@veger 👍 
@dg-spark  ❓ 
@hkirk  ❓ 
@BBBThunda 👍
@pchop2  ❓ 
@arthurlogilab 👍

@JordiCorbilla 👍
